ERCS/erc-7786.md

```diff
@@ -87,27 +102,49 @@ A gateway MAY be upgraded with support for additional attributes. Once present s
 
-MUST revert with `UnsupportedAttribute` if an unsupported attribute key is included. MAY revert if the value of an attribute is not a valid encoding for its expected type.
+MAY accept call value (native token) to be used by attributes or to be sent with the message. In the latter case, it is unspecified how this value is represented on the destination. MUST revert if more call value is included than is needed by attributes and sending value with the message is not a feature supported by the gateway.
 
-MAY accept call value (native token) to be sent with the message. MUST revert if call value is included but it is not a feature supported by the gateway. It is unspecified how this value is represented on the destination.
-
-MAY generate and return a unique non-zero _outbox identifier_, otherwise returning zero. This identifier can be used to track the lifecycle of the message in the outbox in events and for post-processing.
+MAY generate and return a unique non-zero _outbox identifier_, otherwise returning zero. This identifier can be used to track the lifecycle of the message in the outbox in events and for relaying.
 
 MUST emit a `MessagePosted` event, including the optional outbox identifier that is returned by the function.
 
+MAY require further action to have the message relayed to the receiver on destination, in particular paying for the gas cost of delivery. See Relaying.
+
 #### `MessagePosted`
 
 This event signals that a would-be sender has requested a message to be sent.
 
-If `outboxId` is present, post-processing MAY be required to send the message through the cross-chain channel.
+If `outboxId` is present, further action MAY be required. See Relaying.
+
+#### Relaying
+
+The delivery of a message to its receiver will incur a gas cost that the protocol must cover. This includes execution on the destination chain, and MAY include other intermediate processing necessary prior to execution, such as proof generation, verification, etc. The following functions of a gateway allow the sender to pay for this cost.
+
+Note that relaying SHOULD be permissionless. No relayer should be relied on for liveness of the protocol.
+
+#### `quoteRelay`
 
-#### Post-processing
+Returns an estimated amount of native token required to relay a message with the provided parameters and with a certain gas limit.
 
-After a sender has invoked `sendMessage`, further action MAY be required by the gateways to make the message effective. This is called _post-processing_. For example, some payment is typically required to cover the gas of executing the message at the destination.
+#### `requestRelay`
+
+Requests that the message identified by `outboxId` be relayed and delivered to the receiver.
+
+Pays via `msg.value` for the execution of the message on the receiver up to `gasLimit`. A relayer fee MAY be deducted, the specifics of which are unspecified and SHOULD be available out of band (e.g., as gateway documentation).
```

Comment:
This does not support custom base tokens if I understand correctly. Or should the conversion be handled by the protocol? It cannot be handled by native bridges though.